invokeai/app/services/image_storage.py

```diff
@@ -146,28 +160,45 @@ def get(self, image_type: ImageType, image_name: str) -> Image:
     def get_path(
         self, image_type: ImageType, image_name: str, is_thumbnail: bool = False
     ) -> str:
+        # strip out any relative path shenanigans
+        basename = os.path.basename(image_name)
+
         if is_thumbnail:
             path = os.path.join(
-                self.__output_folder, image_type, "thumbnails", image_name
+                self.__output_folder, image_type, "thumbnails", basename
             )
         else:
-            path = os.path.join(self.__output_folder, image_type, image_name)
+            path = os.path.join(self.__output_folder, image_type, basename)
+
         return path
 
-    def save(self, image_type: ImageType, image_name: str, image: Image, metadata: Dict[str, Any] | None = None) -> str:
-        print(metadata)
-        image_subpath = os.path.join(image_type, image_name)
-        self.__pngWriter.save_image_and_prompt_to_png(
-            image, "", image_subpath, metadata
-        )  # TODO: just pass full path to png writer
-        save_thumbnail(
-            image=image,
-            filename=image_name,
-            path=os.path.join(self.__output_folder, image_type, "thumbnails"),
-        )
+    def validate_path(
+        self, path: str
+    ) -> bool:
+        try:
+            os.stat(path)
+            return True
+        except Exception:
+            return False
+
+
+    def save(self, image_type: ImageType, image_name: str, image: Image, metadata: InvokeAIMetadata | None = None) -> Tuple[str, str, int]:
         image_path = self.get_path(image_type, image_name)
+
+        thumbnail_name = get_thumbnail_name(image_name)
```

Comment:
You may want to move these thumbnail-related lines down to the thumbnail saving code to keep the code together.